pep-0722.rst

```diff
@@ -17,140 +17,196 @@ Resolution:
 Abstract
 ========
 
-This PEP further refines `TypeGuards <typeguards_>`__ to
-indicate when negative type narrowing is deemed safe.
-
-[I'd suggest mentioning PEP 647 explicitly here rather than having the opaque link. You can link to a PEP in RST using :pep:`647` ]
-[I think more context is needed here for readers to understand what "negative" means. Maybe one sentence explaining what typeguards currently do and then another about the negative issue.]
+:pep:`647` introduced the concept of ``TypeGuard`` functions which return true
+if their input parameter matches their target type. For example, a function that
+returns ``TypeGuard[str]`` is assumed to return ``true`` if and only if it's
+input parameter is a ``str``. This allows type checkers to narrow types in this
+positive case.
 
+This PEP further refines :pep:`647` by allowing type checkers to narrow types
+even further when a ``TypeGuard`` function returns false.
 
 Motivation
 ==========
 
-`TypeGuards <typeguards_>`__ are used throughout python
-libraries but cannot be used to determine the negative case:
+`TypeGuards <typeguards_>`__ are used throughout Python libraries to allow a
+type checker to narrow the type of something when the ``TypeGuard``
+returns true.
+
+.. code-block:: python
+
+    def is_str(val: str | int) -> TypeGuard[str]:
+        return isinstance(val, str)
+
+    def func(val: str | int):
+        if is_str(val):
+            # Type checkers can assume val is a 'str' in this branch
 
-[Again, more context is needed for the user to understand what "the negative case" means.]
-[Also what does "determine the negative case" mean? Maybe something like "narrow the type in the negative case" would be more clear? Also see the use of that phrase below the code block.]
-[python should be capitalized]
+However, in the ``else`` clause, :pep:`647` didn't prescribe what the type might
+be:
 
-::
-[I'm wondering if `::` is equivalent to `.. code-block:: python` -- You may need the latter to get proper colorization. Check after you build your RST to HTML.]
+.. code-block:: python
 
     def is_str(val: str | int) -> TypeGuard[str]:
         return isinstance(val, str)
 
     def func(val: str | int):
         if is_str(val):
-            reveal_type(val) # str
+            # Type checkers can assume val is a 'str' in this branch
         else:
-            reveal_type(val) # str | int
+            # Type here is not narrowed. It is still 'str | int' 
+
 
-This inability to determine the negative case makes ``TypeGuard`` not as useful as
-it could be.
+This PEP proposes that when the type argument of the ``TypeGuard`` is a subtype
+of the type of the first input argument, then the false case can be further
```

Comment:
IIRC, an argument is a value that you pass to a function, so what you're talking about here is a parameter.

```suggestion
of the type of the first input parameter, then the false case can be further
```